tket/src/Characterisation/Cycles.cpp

```diff
@@ -117,8 +116,7 @@ std::pair<unsigned, std::set<unsigned>> CycleFinder::make_cycle(
     for (const UnitID& uid : this->cycle_history.history[key]) {
       if (not_mergeable_uids.find(uid) != not_mergeable_uids.end()) {
         not_mergeable_keys.insert(key);
-        // break;
-        continue;
+        break;
```

Comment:
Did you mean to add the `break;` here? Seems like a change to the logic?

Reply:
I meant to have the `break` in the first place, the `continue` was checking for a different error that I forgot about. 

Essentially once `key` is known to be "not_mergeable"  we then don't need to check the remaining UnitID. It ends up not being a change to the logic as `not_mergeable_keys` is a Set: if we don't `continue` or don't have the `break` then it will just check extra UnitID to add the same `key`.